bigquery.it.ITNightlyBigQueryTest: testForTableNotFound failed

This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: bbbc4fe5f9da2ddcaaf70277ac0e6f8f37392ed7
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/552b6bfd-15b7-44fb-8239-f26483308b6c), [Sponge](http://sponge2/552b6bfd-15b7-44fb-8239-f26483308b6c)
status: failed
<details><summary>Test output</summary><br><pre>com.google.cloud.bigquery.BigQuerySQLException: Read timed out
	at com.google.cloud.bigquery.it.ITNightlyBigQueryTest.testForTableNotFound(ITNightlyBigQueryTest.java:518)
Caused by: com.google.cloud.bigquery.BigQueryException: Read timed out
	at com.google.cloud.bigquery.it.ITNightlyBigQueryTest.testForTableNotFound(ITNightlyBigQueryTest.java:518)
Caused by: java.net.SocketTimeoutException: Read timed out
	at com.google.cloud.bigquery.it.ITNightlyBigQueryTest.testForTableNotFound(ITNightlyBigQueryTest.java:518)
</pre></details>

This looks like a Socket timeout before we initialise the read API `Caused by: java.net.SocketTimeoutException: Read timed out` . We have recently set `timeoutMs` to 60sec while performing `getQueryResults` call and I guess this is what which might be causing it.

I will add some additional logs to validate if the above hypothesis is true, if yes, then either we will need to decrease the `timeoutMs` or find a way to increase the socket timeout [preferred] (as far as I can see it's not been wired at our RPC layer currently) 

CCing @shollyman JFYI as we recently discussed around this. Ref: https://github.com/googleapis/java-bigquery/pull/2222

--- a/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ConnectionImpl.java
+++ b/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ConnectionImpl.java
@@ -190,11 +190,13 @@ class ConnectionImpl implements Connection {
     try {
       // use jobs.query if all the properties of connectionSettings are supported
       if (isFastQuerySupported()) {
+        logger.log(Level.INFO, "\n Using Fast Query Path");
         String projectId = bigQueryOptions.getProjectId();
         QueryRequest queryRequest = createQueryRequest(connectionSettings, sql, null, null);
         return queryRpc(projectId, queryRequest, sql, false);
       }
       // use jobs.insert otherwise
+      logger.log(Level.INFO, "\n Not Using Fast Query Path, using jobs.insert");
       com.google.api.services.bigquery.model.Job queryJob =
           createQueryJob(sql, connectionSettings, null, null);
       JobId jobId = JobId.fromPb(queryJob.getJobReference());
@@ -233,12 +235,14 @@ class ConnectionImpl implements Connection {
     try {
       // use jobs.query if possible
       if (isFastQuerySupported()) {
+        logger.log(Level.INFO, "\n Using Fast Query Path");
         final String projectId = bigQueryOptions.getProjectId();
         final QueryRequest queryRequest =
             createQueryRequest(connectionSettings, sql, parameters, labelMap);
         return queryRpc(projectId, queryRequest, sql, parameters != null);
       }
       // use jobs.insert otherwise
+      logger.log(Level.INFO, "\n Not Using Fast Query Path, using jobs.insert");
       com.google.api.services.bigquery.model.Job queryJob =
           createQueryJob(sql, connectionSettings, parameters, labelMap);
       JobId jobId = JobId.fromPb(queryJob.getJobReference());
@@ -1003,8 +1007,7 @@ class ConnectionImpl implements Connection {
     // progress (a job won't get stuck in pending forever).
     boolean jobComplete = false;
     GetQueryResultsResponse results = null;
-    long timeoutMs =
-        60000; // defaulting to 60seconds. TODO(prashant): It should be made user configurable
+    long timeoutMs = 10000; // defaulting to 10seconds.
 
     while (!jobComplete) {
       try {
@@ -1032,6 +1035,7 @@ class ConnectionImpl implements Connection {
           throw new BigQueryException(bigQueryErrors);
         }
       } catch (BigQueryRetryHelper.BigQueryRetryHelperException e) {
+        logger.log(Level.WARNING, "\n Error occurred while calling getQueryResultsWithRowLimit", e);
         throw BigQueryException.translateAndThrow(e);
       }
       jobComplete = results.getJobComplete();
@@ -1285,8 +1289,10 @@ class ConnectionImpl implements Connection {
               bigQueryOptions.getClock(),
               retryConfig);
     } catch (BigQueryRetryHelper.BigQueryRetryHelperException e) {
+      logger.log(Level.WARNING, "\n Error occurred while calling createJobForQuery", e);
       throw BigQueryException.translateAndThrow(e);
     }
+    logger.log(Level.INFO, "\n Query job created");
     return queryJob;
   }
 

--- a/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITNightlyBigQueryTest.java
+++ b/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITNightlyBigQueryTest.java
@@ -496,7 +496,7 @@ public class ITNightlyBigQueryTest {
     String query =
         String.format(
             "SELECT * FROM `bigquery-samples.wikipedia_benchmark.Wiki10B` LIMIT %s", recordCnt);
-
+    logger.log(Level.INFO, "Query used: {0}", query);
     String dataSet = RemoteBigQueryHelper.generateDatasetName();
     String table = "TAB_" + UUID.randomUUID();
     createDataset(dataSet);

